crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -44,7 +45,7 @@ impl<'w> EntityRef<'w> {
     }
 
     #[inline]
-    pub fn world(&mut self) -> &World {
+    pub fn world(self) -> &'w World {
```

Comment:
Why is the lifetime explicit here but not below in the `EntityMut` equivalent?

Reply:
`EntityMut` ties a borrow to `&self` wheras this doesnt